### PR TITLE
refactor(fcp): Detach GET runtime handoff

### PR DESCRIPTION
**Pull request by leumor**
## Summary
- complete the first FCP-side GET runtime handoff for PR-172 by replacing live fetch-runtime types in `:adapter-fcp` with adapter-owned detached config and execution seams
- move concrete GET execution assembly, fetch-context translation, restore helpers, and Binary Blob bucket handling into `:bridge-fcp-runtime`
- update GET-path tests, boundary coverage, and Javadoc for the new seam classes while preserving FCP wire behavior and persisted request format

## How to test
- `./gradlew :adapter-fcp:test --tests network.crypta.clients.fcp.AdapterFcpBoundaryTest`
- `./gradlew :bridge-fcp-runtime:test --tests network.crypta.clients.fcp.bridge.CoreFcpFetchRuntimeSupportTest --tests network.crypta.clients.fcp.bridge.CoreFcpServerDependenciesFactoryTest --tests network.crypta.clients.fcp.bridge.FcpPersistentRequestRecoveryCodecTest`
- `./gradlew :test --tests network.crypta.clients.fcp.ClientGetFactoryTest --tests network.crypta.clients.fcp.ClientGetGetterFactoryTest --tests network.crypta.clients.fcp.ClientGetPersistenceCodecTest --tests network.crypta.clients.fcp.ClientGetPersistenceIOTest --tests network.crypta.clients.fcp.ClientGetRestartCoordinatorTest --tests network.crypta.clients.fcp.ClientGetLifecycleTest --tests network.crypta.clients.fcp.ClientGetStatusSnapshotBuilderTest --tests network.crypta.clients.fcp.ClientGetTest --tests network.crypta.clients.fcp.ClientGetMessageReplayTest --tests network.crypta.clients.fcp.FCPServerTest --tests network.crypta.clients.fcp.ClientGetFetchConfigTest`
- `./gradlew test`

### CODE REVIEW

File: adapter-fcp/src/main/java/network/crypta/clients/fcp/ClientGetPersistenceCodec.java [hunk 1]
```diff
@@ -106,7 +107,7 @@ static void writeClientDetail(ClientGet request, DataOutputStream dos, ChecksumC
     }
     dos.writeBoolean(request.binaryBlobRequested());
     try (DataOutputStream ctxStream = ClientGetGetterFactory.checksummedWriter(dos, checker)) {
-      request.fetchContextForGetter().writeTo(ctxStream);
+      request.runtimeFetchSupport().encodeFetchConfig(request.fetchConfig(), ctxStream);
```
[the remark this answers is not in the document]
Comment (leumor) [reply adapter-fcp/src/main/java/network/crypta/clients/fcp/ClientGetPersistenceCodec.java]: Fixed in `8c1e503556`.

`ClientGet` now caches the canonical fetch-config encoding when it is Java-serialized, and `ClientGetPersistenceCodec.writeClientDetail()` falls back to that cached encoding when the transient `runtimeFetchSupport` is unavailable on a deserialized request. When a live connection is present, it also rehydrates the seam from the current server before encoding.

That keeps recovery-data checkpoints writable for serialized-resume GETs instead of throwing a `NullPointerException`.

Validation:
- `./gradlew :test --tests network.crypta.clients.fcp.ClientGetPersistenceCodecTest`
- `./gradlew :test --tests network.crypta.clients.fcp.ClientGetPersistenceCodecTest --tests network.crypta.clients.fcp.ClientGetPersistenceIOTest --tests network.crypta.clients.fcp.ClientGetMessageReplayTest --tests network.crypta.clients.fcp.FCPServerTest :bridge-fcp-runtime:test --tests network.crypta.clients.fcp.bridge.FcpPersistentRequestRecoveryCodecTest`

File: adapter-fcp/src/main/java/network/crypta/clients/fcp/ClientGet.java [hunk 2]
```diff
@@ -626,14 +634,14 @@ File targetFileForLifecycle() {
...
-   * treated as read-only by callers outside the request lifecycle.
+   * tracking. The returned object is the request's active requester and should be treated as
+   * read-only by callers outside the request lifecycle.
    *
    * @return client requester instance used to run this request.
    */
   @Override
   protected ClientRequester getClientRequest() {
-    return getter;
+    return execution == null ? null : execution.requester();
```
[the remark this answers is not in the document]
Comment (leumor) [reply adapter-fcp/src/main/java/network/crypta/clients/fcp/ClientGet.java:644]: Fixed in `5e60c55e3d`.

`ClientGet` no longer relies solely on a live connection-local fetch seam during resume. The forever-persistent root is now wired to the server fetch runtime support, `ClientGet.runtimeFetchSupport()` resolves through that root when the connection is absent, and serialized GET executions now rebind their transient runtime context on `onResume(...)`. For older serialized snapshots that still have `execution == null`, `ClientGet.innerResume()` rebuilds a fresh execution from the detached request state and starts it instead of leaving the request permanently stalled.

Validation:
- `./gradlew :test --tests network.crypta.clients.fcp.ClientGetTest --tests network.crypta.clients.fcp.ClientGetPersistenceCodecTest --tests network.crypta.clients.fcp.FCPServerTest --tests network.crypta.clients.fcp.PersistentRequestRootTest`
- `./gradlew :bridge-fcp-runtime:test --tests network.crypta.clients.fcp.bridge.CoreFcpFetchRuntimeSupportTest`

File: adapter-fcp/src/main/java/network/crypta/clients/fcp/ClientGetPersistenceCodec.java [hunk 3]
```diff
@@ -283,6 +288,40 @@ private static ClientGet.ReturnType parseReturnType(short code) throws StorageFo
...
+        byte[] encodedFetchConfig = buffer.toByteArray();
+        request.setPersistedFetchConfigEncoding(encodedFetchConfig);
+        return encodedFetchConfig;
+      }
+    }
+    byte[] cachedEncoding = request.persistedFetchConfigEncoding();
+    if (cachedEncoding != null) {
+      return cachedEncoding;
+    }
+    throw new IOException("No runtime fetch support or cached fetch configuration encoding");
```
[the remark this answers is not in the document]
Comment (leumor) [reply adapter-fcp/src/main/java/network/crypta/clients/fcp/ClientGetPersistenceCodec.java:306]: Fixed in `5e60c55e3d`.

Checkpoint persistence no longer depends on an active connection or on the cached fetch-config bytes being present. The forever-persistent root now keeps the server-owned `FcpFetchRuntimeSupport`, and `ClientGet.runtimeFetchSupport()` resolves through that root when a deserialized request has no live connection. That gives `writeClientDetail()` a fetch-config encoder during startup/disconnect windows, including requests restored from older serialized snapshots that predate `persistedFetchConfigEncoding`.

I also added a regression test that exercises `writeClientDetail()` with `runtimeFetchSupport == null`, no cached bytes, and only the persistent root wiring available.

Validation:
- `./gradlew :test --tests network.crypta.clients.fcp.ClientGetTest --tests network.crypta.clients.fcp.ClientGetPersistenceCodecTest --tests network.crypta.clients.fcp.FCPServerTest --tests network.crypta.clients.fcp.PersistentRequestRootTest`
- `./gradlew :bridge-fcp-runtime:test --tests network.crypta.clients.fcp.bridge.CoreFcpFetchRuntimeSupportTest`
